src/bin/cargo-flamegraph.rs

```diff
@@ -117,7 +117,20 @@ enum Opts {
 
 fn build(opt: &Opt) {
     let mut cmd = std::process::Command::new("cargo");
-    cmd.arg("build");
+
+    if opt.bench.is_none() || opt.dev {
+        cmd.arg("build");
+    } else {
+        // This will build benchmarks with the `bench` profile. This is needed
+        // because the `--profile` argument for `cargo build` is unstable.
+        cmd.arg("bench");
+        cmd.arg("--no-run");
+    }
+
+    if let Some(ref bench) = opt.bench {
+        cmd.arg("--bench");
+        cmd.arg(bench);
+    }
```

Comment:
Would you mind squashing the commits and leaving the `if let Some(ref bench) = opt.bench {}` is its former place? That would make for a nicely minimal change which I think solves your issue.

Reply:
Did that. Slightly rewrote the whole thing actually.